packages/launch/src/hooks/use-on-launch.ts

```diff
@@ -72,5 +62,5 @@ export const useOnLaunch = () => {
 			}
 			window.top.location.href = `https://wordpress.com/home/${ siteId }`;
 		}
-	}, [ launchStatus ] );
+	}, [ isLaunched ] ); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Is there a reason why we're not setting all of the correct dependencies for this `useEffect` hook?

Reply:
We want to run the code inside only once, when `isLaunched` becomes `true`. So think of it like `launchSite.then( () => { /* run what's inside the if( isLaunched ) condition above */ } )`.

I know the implementation with `useEffect` for this is not stellar and some chained action creators on Launch store would've been probably close to ideal. But because of the way wp-data works (or how we're using it) we ended up here. Anyway, once we migrate Step-by-step launch flow to using Editor checkout, we can remove this hook.

Reply:
Thank you for the explanation, good to keep this in mind for the next time we need to implement this sort of logic.

For the purpose of this PR, we can consider this conversation resolved